### DISCUSSION TITLE
"jvm-abi-gen.jar" Error while building the project with Bazel

### DISCUSSION THREAD

**Discussion by sandy99405** [oppia/oppia-android#5300] [General Discussion]
I'm facing an error while running Bazel. 
I was trying to run the following commands from D:/git/oppia-android  folder
```````` 
set REPIN=1 bazel run @unpinned_maven//:pin

bazel run //scripts:generate_maven_dependencies_list -- $(pwd) third_party/maven_install.json scripts/assets/maven_dependencies.textproto scripts/assets/maven_dependencies.pb
`````````

![Screenshot (2384)](https://github.com/oppia/oppia-android/assets/153572065/2c3710da-32d9-4755-ad61-d6cc758800a4)

but it's building the bazel output in : 
```````
  C:\Users\ASUS\_bazel_ASUS\uzib7o6s\execroot 
`````````

![Screenshot (2380)](https://github.com/oppia/oppia-android/assets/153572065/9989e760-b624-47ef-8c57-cbe74283ac45)

it says that the following .jar is missing
```````
external/com_github_jetbrains_kotlin/lib/jvm-abi-gen.jar
```````
![Screenshot (2383)](https://github.com/oppia/oppia-android/assets/153572065/de949894-a5ba-4b46-9d08-26139e8385b1)

in the execroot folder, the build is creating a file with the name DO_NOT_BUILD_HERE. which has the content as : D/git/oppia-android.
(Which is where I ran the commands from)

** @Rd4dev  or  @theMr17  or anyone Please help me with this error if you have encountered it before**.🙏

**Comment by Rd4dev** [comment]
Hi @sandy99405, I've been using the Gradle build for my development. However, I recently installed Bazel during a [discussion](https://github.com/oppia/oppia-android/discussions/5241). I believe I can give the Bazel build a shot and contribute positively. I'll keep you posted on my progress soon.

**Reply by sandy99405** [reply]
**Thanks a million** @Rd4dev 

I've been trying and solving errors just to get a new error 🥲.......I've installed Bazel in many different ways but failed terribly.

I have few questions if you don't mind

1) Where do I install the Bazel correctly ?

2) My aim is to run this command successfully (not bazel build //:oppia)
````````
bazel run //scripts:generate_maven_dependencies_list -- $(pwd) third_party/maven_install.json scripts/assets/maven_dependencies.textproto scripts/assets/maven_dependencies.pb
````````
3) it's unable to fetch the dependencies from the links:

for Ex: 
````````````````
WARNING: Download from https://maven.google.com/com/squareup/moshi/moshi-kotlin-codegen/1.11.0/moshi-kotlin-codegen-1.11.0.jar failed: class com.google.devtools.build.lib.bazel.repository.downloader.UnrecoverableHttpException GET returned 404 Not Found

````````````````
even when I open it physically, it's showing error not found in my Browser.(I'm from India btw) should I use a VPN for it?

**Reply by adhiamboperes** [reply]
@MohitGupta121, could you please confirm if VPN is needed here, per your previous comments?

**Reply by MohitGupta121** [reply]
@sandy99405 which company wifi connection you using? If you using jio then I prefer to use vpn. Thanks!

**Reply by sandy99405** [reply]
Yes @MohitGupta121  I've been using Jio fiber net.......and I've also tried VPN of U.S but it didn't work

**Reply by sandy99405** [reply]
@MohitGupta121  can you please confirm which country VPN works ?

**Reply by MohitGupta121** [reply]
Use any random vpn on your computer, for me in mac, I used Turbo vpn.

**Reply by sandy99405** [reply]
ok thank you 😊....I will try and update

**Reply by sandy99405** [reply]
Thank you all very much for the help guys. The problem is resolved now ✌️

**Comment by adhiamboperes** [comment]
@sandy99405, could you please add your solution as a comment on this thread?

**Reply by sandy99405** [reply]
@adhiamboperes The problem was that I did it directly in Windows Command Prompt which Oppia did not recommend. I solved it by using WSL.
